### PR TITLE
Removed unused variables, add some constants numbers

### PR DESCRIPTION
**Pull request by qarmin**
I added check `storage->frame.current_rt` in rasterizer_scene_gles2.cpp, because this variable was checked later in code e.g. line 3319 or 3335

Also there are also new return checked from function unzGoToNextFile

0.7071 was changed to Math_SQRT12

Changes in text_edit are caused because in Valgrind I got more than hundred reports about using uninitialised variables
```

==18375== Conditional jump or move depends on uninitialised value(s)
==18375==    at 0x2D38FBB: TextEdit::times_line_wraps(int) const (text_edit.cpp:4327)
==18375==    by 0x2D39EDE: TextEdit::cursor_set_line(int, bool, bool, int) (text_edit.cpp:4491)
==18375==    by 0x2D3B6CA: TextEdit::set_text(String) (text_edit.cpp:4801)
==18375==    by 0x27D61DC: VersionControlEditorPlugin::VersionControlEditorPlugin() (version_control_editor_plugin.cpp:534)
==18375==    by 0x27D194A: VersionControlEditorPlugin::get_singleton() (version_control_editor_plugin.cpp:83)
==18375==    by 0x217B00D: EditorNode::EditorNode() (editor_node.cpp:6074)
==18375==    by 0x14251D3: Main::start() (main.cpp:1677)
==18375==    by 0x13E4957: main (godot_x11.cpp:55)
==18375== 
==18375== Conditional jump or move depends on uninitialised value(s)
==18375==    at 0x2D3AA0B: TextEdit::get_column_x_offset_for_line(int, int) const (text_edit.cpp:4645)
==18375==    by 0x2D39937: TextEdit::cursor_set_column(int, bool) (text_edit.cpp:4449)
==18375==    by 0x2D3B6E0: TextEdit::set_text(String) (text_edit.cpp:4802)
==18375==    by 0x27D61DC: VersionControlEditorPlugin::VersionControlEditorPlugin() (version_control_editor_plugin.cpp:534)
==18375==    by 0x27D194A: VersionControlEditorPlugin::get_singleton() (version_control_editor_plugin.cpp:83)
==18375==    by 0x217B00D: EditorNode::EditorNode() (editor_node.cpp:6074)
==18375==    by 0x14251D3: Main::start() (main.cpp:1677)
==18375==    by 0x13E4957: main (godot_x11.cpp:55)
```

### CODE REVIEW

File: core/io/file_access_zip.cpp [hunk 1]
```diff
@@ -200,7 +200,7 @@ bool ZipArchive::try_open_pack(const String &p_path, bool p_replace_files) {
 
 		unz_file_info64 file_info;
 		err = unzGetCurrentFileInfo64(zfile, &file_info, filename_inzip, sizeof(filename_inzip), NULL, 0, NULL, 0);
-		ERR_CONTINUE(err != UNZ_OK);
+		ERR_CONTINUE(err != UNZ_EOF && err != UNZ_OK);
```
Comment (akien-mga) [inline core/io/file_access_zip.cpp]: Was this prompted by a warning? It may change the function logic.

File: drivers/gles2/rasterizer_scene_gles2.cpp [hunk 2]
```diff
@@ -3300,7 +3300,7 @@ void RasterizerSceneGLES2::render_scene(const Transform &p_cam_transform, const
 		reflection_probe_count = 0;
 	}
 
-	if (env && env->bg_mode == VS::ENV_BG_CANVAS) {
+	if (storage->frame.current_rt && env && env->bg_mode == VS::ENV_BG_CANVAS) {
```
Comment (akien-mga) [inline drivers/gles2/rasterizer_scene_gles2.cpp]: Why is this needed here and not before? There's already code a few lines above that dereferences this pointer.

File: editor/editor_asset_installer.cpp [hunk 3]
```diff
@@ -107,6 +107,7 @@ void EditorAssetInstaller::open(const String &p_path, int p_depth) {
 		files_sorted.insert(name);
 
 		ret = unzGoToNextFile(pkg);
+		ERR_BREAK(ret != UNZ_EOF && ret != UNZ_OK);
```
Comment (akien-mga) [inline editor/editor_asset_installer.cpp]: This defeats the purpose of the conditional in the `while` loop. What warning is this handling?

File: editor/editor_asset_installer.cpp [hunk 4]
```diff
@@ -281,6 +282,7 @@ void EditorAssetInstaller::ok_pressed() {
 
 		idx++;
 		ret = unzGoToNextFile(pkg);
+		ERR_BREAK(ret != UNZ_EOF && ret != UNZ_OK);
```
Comment (akien-mga) [inline editor/editor_asset_installer.cpp]: Same here.

File: editor/export_template_manager.cpp [hunk 5]
```diff
@@ -236,6 +236,7 @@ bool ExportTemplateManager::_install_from_file(const String &p_file, bool p_use_
 		}
 
 		ret = unzGoToNextFile(pkg);
+		ERR_BREAK(ret != UNZ_EOF && ret != UNZ_OK);
```
Comment (akien-mga) [inline editor/export_template_manager.cpp]: I'd prefer the unzip changes moved to their own PR, with more explanations on what prompted the change and what it solves.

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 6]
```diff
@@ -54,7 +54,7 @@
 #define MIN_ZOOM 0.01
 #define MAX_ZOOM 100
 
-#define RULER_WIDTH 15 * EDSCALE
+#define RULER_WIDTH (15 * (EDSCALE))
```
Comment (akien-mga) [inline editor/plugins/canvas_item_editor_plugin.cpp]: That's a bit overkill, why parentheses around `EDSCALE`?

File: modules/assimp/editor_scene_importer_assimp.cpp [hunk 7]
```diff
@@ -393,7 +393,7 @@ EditorSceneImporterAssimp::_generate_scene(const String &p_path, aiScene *scene,
 				ERR_FAIL_COND_V_MSG(parent_node == NULL, state.root,
 						"Parent node invalid even though lookup successful, out of ram?")
 
-				if (parent_node && spatial != state.root) {
+				if (spatial != state.root) {
 					parent_node->add_child(spatial);
 					spatial->set_owner(state.root);
 				} else if (spatial == state.root) {
```
Comment (akien-mga) [inline modules/assimp/editor_scene_importer_assimp.cpp]: Next run of your static analyzer it will complain about this, so best fix it now too (the `else if` should simply be `else`, and the current `else` branch is dead code).

File: core/io/file_access_zip.cpp (hunk 1 above)
Comment (qarmin) [reply core/io/file_access_zip.cpp]: I was looking only at one place with macros, but minizip seems to use also macros from zlib, so I just removed this, because it would actually change the logic of program.

File: modules/gdscript/language_server/gdscript_text_document.cpp [hunk 8]
```diff
@@ -281,8 +281,6 @@ Dictionary GDScriptTextDocument::resolve(const Dictionary &p_params) {
 }
 
 Array GDScriptTextDocument::foldingRange(const Dictionary &p_params) {
-	Dictionary params = p_params["textDocument"];
-	String path = params["uri"];
```
Comment (akien-mga) [inline modules/gdscript/language_server/gdscript_text_document.cpp]: CC @Geequlim. Is it expected that this method returns an empty Array?

Comment (Geequlim) [reply modules/gdscript/language_server/gdscript_text_document.cpp]: @akien-mga Looks good to me.
FoldingRange is implemented in script languge clients so we just return an empty Array here.

Comment (akien-mga) [reply modules/gdscript/language_server/gdscript_text_document.cpp]: Thanks!

File: drivers/gles2/rasterizer_scene_gles2.cpp [hunk 9]
```diff
@@ -3223,7 +3223,7 @@ void RasterizerSceneGLES2::render_scene(const Transform &p_cam_transform, const
 
 		probe_interior = probe->probe_ptr->interior;
 
-	} else {
+	} else if (storage->frame.current_rt) {
```
Comment (akien-mga) [inline drivers/gles2/rasterizer_scene_gles2.cpp]:
```
drivers/gles2/rasterizer_scene_gles2.cpp:3226:13: error: variable

      'viewport_width' is used uninitialized whenever 'if' condition is false

      [-Werror,-Wsometimes-uninitialized]

        } else if (storage->frame.current_rt) {

                   ^~~~~~~~~~~~~~~~~~~~~~~~~

drivers/gles2/rasterizer_scene_gles2.cpp:3249:26: note: uninitialized use occurs

      here

        state.viewport_size.x = viewport_width;

                                ^~~~~~~~~~~~~~

drivers/gles2/rasterizer_scene_gles2.cpp:3226:9: note: remove the 'if' if its

      condition is always true

        } else if (storage->frame.current_rt) {

               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

drivers/gles2/rasterizer_scene_gles2.cpp:3199:20: note: initialize the variable

      'viewport_width' to silence this warning

        int viewport_width, viewport_height;

                          ^

                           = 0

drivers/gles2/rasterizer_scene_gles2.cpp:3226:13: error: variable

      'viewport_height' is used uninitialized whenever 'if' condition is false

      [-Werror,-Wsometimes-uninitialized]

        } else if (storage->frame.current_rt) {

                   ^~~~~~~~~~~~~~~~~~~~~~~~~

drivers/gles2/rasterizer_scene_gles2.cpp:3250:26: note: uninitialized use occurs

      here

        state.viewport_size.y = viewport_height;

                                ^~~~~~~~~~~~~~~

drivers/gles2/rasterizer_scene_gles2.cpp:3226:9: note: remove the 'if' if its

      condition is always true

        } else if (storage->frame.current_rt) {

               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

drivers/gles2/rasterizer_scene_gles2.cpp:3199:37: note: initialize the variable

      'viewport_height' to silence this warning

        int viewport_width, viewport_height;

                                           ^

                                            = 0

2 errors generated.
```

I still don't see the point in adding these checks, as `storage->frame.current_rt` is already used in all the code before these checks. What was the warning?

Comment (akien-mga) [reply drivers/gles2/rasterizer_scene_gles2.cpp]: If you can remove these changes (and maybe open another PR / issue to discuss what the actual warning was), the rest seems good to merge.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

File: modules/recast/navigation_mesh_generator.cpp [hunk 10]
```diff
@@ -62,7 +62,7 @@ void EditorNavigationMeshGenerator::_add_vertex(const Vector3 &p_vec3, Vector<fl
 }
 
 void EditorNavigationMeshGenerator::_add_mesh(const Ref<Mesh> &p_mesh, const Transform &p_xform, Vector<float> &p_verticies, Vector<int> &p_indices) {
-	int current_vertex_count = 0;
```
Comment (aaronfranke) [inline modules/recast/navigation_mesh_generator.cpp]: Isn't it better to have variables start initialized?
